src/test/java/appium/StepDefs.java

```diff
@@ -1,78 +1,125 @@
 package appium;
 
-import io.appium.java_client.MobileBy;
+import io.appium.java_client.AppiumDriver;
+import io.appium.java_client.MobileElement;
 import io.appium.java_client.android.AndroidDriver;
-import io.appium.java_client.android.AndroidElement;
+import io.appium.java_client.ios.IOSDriver;
 import io.cucumber.java.After;
 import io.cucumber.java.Before;
 import io.cucumber.java.Scenario;
 import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
+import org.openqa.selenium.By;
 import org.openqa.selenium.JavascriptExecutor;
 import org.openqa.selenium.remote.DesiredCapabilities;
-import org.openqa.selenium.support.ui.ExpectedConditions;
-import org.openqa.selenium.support.ui.WebDriverWait;
-import org.testng.Assert;
 
 
+import java.io.*;
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.util.List;
+import java.util.Properties;
 import java.util.concurrent.TimeUnit;
 
-public class StepDefs{
-    public AndroidDriver<AndroidElement> driver;
-    DesiredCapabilities caps;
+public class StepDefs {
 
+    DesiredCapabilities caps;
     public final String USERNAME = System.getenv("BROWSERSTACK_USERNAME");
     public final String ACCESS_KEY = System.getenv("BROWSERSTACK_ACCESS_KEY");
     public final String APP_ID = System.getenv("BROWSERSTACK_APP_ID");
     public final String URL = "https://" + USERNAME + ":" + ACCESS_KEY + "@hub-cloud.browserstack.com/wd/hub";
     public JavascriptExecutor jse;
+    public AppiumDriver<MobileElement> driver;
 
     @Before
     public void setup() throws Exception{
         caps = new DesiredCapabilities();
-        caps.setCapability("device", "Samsung Galaxy S9");
-        caps.setCapability("os_version", "8.0");
-        caps.setCapability("name", "parallel_test");
-        caps.setCapability("build", "browserstack-appium-java-cucumber-example");
-        if(APP_ID!=null){
-            System.out.println("Setting from Environment variable");
-            caps.setCapability("app", APP_ID);
+
+        if(readPropertiesFile("platformName").equals("Android"))
+        {
+            caps.setCapability("deviceName", readPropertiesFile("android_device"));
+            caps.setCapability("osVersion", readPropertiesFile("android_os_version"));
+            caps.setCapability("name", readPropertiesFile("android_testName")+" "+readPropertiesFile("android_device"));
+            caps.setCapability("build", readPropertiesFile("android_build"));
+            if(APP_ID!=null){
+                System.out.println("Setting from Environment variable");
+                caps.setCapability("app", "bs://96c11ded5deb147ec2e6977c1a940d1f0f17e14a");
+            }
+            else
+                caps.setCapability("app", "browserStackDemoApp");
```

Comment:
This wont work for end user as they wont be using this specific custom id. Lets either define a variable for custom id or remove this else condition.